File: tools/python-incident-analyzer/src/incident_analyzer/domain/incident_stats.py
"""Incident statistics domain model."""

from __future__ import annotations

from dataclasses import dataclass
from typing import Any


@dataclass(frozen=True)
class IncidentStats:
    """Aggregated incident counters returned by the Java backend."""

    total: int
    open: int
    investigating: int
    resolved: int
    critical: int
    high: int
    medium: int
    low: int

    @classmethod
    def from_json(cls, data: dict[str, Any]) -> IncidentStats:
        """Create incident statistics from backend JSON."""
        return cls(
            total=int(data.get("total", 0)),
            open=int(data.get("open", 0)),
            investigating=int(data.get("investigating", 0)),
            resolved=int(data.get("resolved", 0)),
            critical=int(data.get("critical", 0)),
            high=int(data.get("high", 0)),
            medium=int(data.get("medium", 0)),
            low=int(data.get("low", 0)),
        )

File: frontend/flutter-rules-practice/lib/features/incidents/domain/incident_stats.dart
class IncidentStats {
  final int total;
  final int open;
  final int investigating;
  final int resolved;
  final int low;
  final int medium;
  final int high;
  final int critical;

  const IncidentStats({
    required this.total,
    required this.open,
    required this.investigating,
    required this.resolved,
    required this.low,
    required this.medium,
    required this.high,
    required this.critical,
  });

  factory IncidentStats.fromJson(Map<String, dynamic> json) {
    return IncidentStats(
      total: json['total'] as int,
      open: json['open'] as int,
      investigating: json['investigating'] as int,
      resolved: json['resolved'] as int,
      low: json['low'] as int,
      medium: json['medium'] as int,
      high: json['high'] as int,
      critical: json['critical'] as int,
    );
  }
}